### PR TITLE
Fix a bug in Khuri-Makdisi small model of Jacobian

### LINKED ISSUE #39148: Jacobian arithmetic assertion error km-small model
### Steps To Reproduce

While doing some testing of the `km-small` Jacobian model, I ran into this example which resulted in a failed assertion error when trying to add two elements.

```
k.<x> = FunctionField(GF(17))
t = polygen(k)

F.<y> = k.extension(t^4 + (14 * x + 14) * t^3 + 9 * t^2 + (10 * x^2 + 15 * x + 8) * t + 7 * x^3 + 15 * x^2 + 6 * x + 16)
x = F(x)
O = F.maximal_order()

infty1, infty2 = F.places_infinite()

P = O.ideal((x + 1, y + 7)).divisor()
D1 = 3 * infty2 + infty1 - 4 * P
D2 = F.divisor_group().zero()

# These are both Jacobian elements
assert D1.degree() == 0
assert D2.degree() == 0

J = F.jacobian(model='km-small', base_div = 4 * P)

JD1 = J(D1)
JD2 = J(D2)

JD1 + JD2  # AssertionError 
```

### Expected Behavior

Adding any two elements of the Jacobian should give a result

### Actual Behavior

Adding these two elements gives an error

### Additional Information

Tagging @kwankyu who added the Khuri-Makdisi Jacobian model in #35467.

The base divisor in the above example is `4*Place (x + 1, y + 7)`.

Additional pairs of elements that give an error when trying to add (same Jacobian as in the example above):
```
D1: 3*Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 4*Place (x + 1, y + 7)
D2: 0

D1: 3*Place (1/x, 1/x*y)
 - 3*Place (x + 1, y + 7)
D2: Place (1/x, 1/x*y + 14)
 - Place (x + 1, y + 7)

D1: 2*Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 3*Place (x + 1, y + 7)
D2: Place (1/x, 1/x*y)
 - Place (x + 1, y + 7)

D1: 2*Place (1/x, 1/x*y)
 - 2*Place (x + 1, y + 7)
D2: Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 2*Place (x + 1, y + 7)

D1: Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 2*Place (x + 1, y + 7)
D2: 2*Place (1/x, 1/x*y)
 - 2*Place (x + 1, y + 7)

D1: Place (1/x, 1/x*y)
 - Place (x + 1, y + 7)
D2: 2*Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 3*Place (x + 1, y + 7)

D1: Place (1/x, 1/x*y + 14)
 - Place (x + 1, y + 7)
D2: 3*Place (1/x, 1/x*y)
 - 3*Place (x + 1, y + 7)

D1: 0
D2: 3*Place (1/x, 1/x*y)
 + Place (1/x, 1/x*y + 14)
 - 4*Place (x + 1, y + 7)
```

### Environment

- **OS**: Fedora 41
- **Sage Version**: 10.5

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by kwankyu**
Fixes #39148.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/function_field/khuri_makdisi.pyx [hunk 1]
```diff
@@ -871,13 +871,31 @@ cdef class KhuriMakdisi_small(KhuriMakdisi_base):
...
+            sage: k.<x> = FunctionField(GF(17)); t = polygen(k)
+            sage: F.<y> = k.extension(t^4 + (14*x + 14)*t^3 + 9*t^2 + (10*x^2 + 15*x + 8)*t
+            ....:  + 7*x^3 + 15*x^2 + 6*x + 16)
+            sage: infty1, infty2 = F.places_infinite()
+            sage: O = F.maximal_order()
+            sage: P = O.ideal((x + 1, y + 7)).divisor()
+            sage: D1 = 3*infty2 + infty1 - 4*P
+            sage: D2 = F.divisor_group().zero()
+            sage: J = F.jacobian(model='km-small', base_div=4*P)
+            sage: J(D1) + J(D2) == J(D1)
```
Comment (vincentmacri) [inline src/sage/rings/function_field/khuri_makdisi.pyx:887]:
```suggestion
            sage: J(D1) + J(D2) == J(D1)  # long time
```
I think this should get rid of the warning (and correctly mark this as a slow test)

**Comment by vincentmacri** [comment]
Looks good! Feel free to set this to positive review after adding the `# long time` marker to the test.

File: src/sage/rings/function_field/khuri_makdisi.pyx (hunk 1 above)
Comment (kwankyu) [reply src/sage/rings/function_field/khuri_makdisi.pyx:887]: No. The block tag `sage: # long time` at line 877 already marks this line as a slow test and correctly skipped. But doctesting in github ci runs with `--long --warn-long 30.0`. Hence we could get the slow test warning anyway even with `# long time` tag.

Comment (vincentmacri) [reply src/sage/rings/function_field/khuri_makdisi.pyx:887]: Good to know. Everything is good then.

**Comment by kwankyu** [comment]
Thanks for review!

**Pull request merged**